src/python/pants/backend/python/util_rules/local_dists.py

```diff
@@ -178,6 +178,12 @@ async def build_local_dists(
         ),
     )
 
+    if not wheels:
```

Comment:
Should we also shortcircuit at line 146? Other than an extra 2 lines of code, I see no reason not to.

Also I'm not sure this informative comment is necessary - short circuiting usually seems like the right thing to do!

Reply:
> Should we also shortcircuit at line 146? Other than an extra 2 lines of code, I see no reason not to.

That would require adjusting the signature of `LocalDistsPex` to make the pex optional: I started down that path, but shortcircuiting here allowed for the null-object pattern.

> Also I'm not sure this informative comment is necessary - short circuiting usually seems like the right thing to do!

Yeaaaa. @benjyw and I both felt that landing this was punting on determining what the actual performance issue was in the trailing code: he's following up our best guess there. If he determines why then he can delete the comment perhaps, but in the meantime let's leave it.